### ISSUE TITLE
Accordion Block: Add and dynamically manage region roles

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#71779]
### Description

The accordion content panel has an `aria-labelledby` attribute that points to the panel header. It does not, however, have a corresponding `role`. Without a `role` attribute, the aria label will not be announced.

The content should have `role="region"` to ensure this label is announced.

However, the WAI pattern for an accordion recommends only doing this for short accordions or when the accordion content is complex; specifically when it contains headings.

If possible, it *might* be helpful to dynamically manage the presence of `role="region"`, so that it is applied if the content of the accordion content container contains headings or if there are a small number of accordion items. (The WAI pattern suggests 6 or fewer.)

I'm open to discussion on the specifics there; personally, I think it can potentially make things more confusing if some accordions use `role="region"` and others don't. That makes for an inconsistent user experience - for example, if a site has several pages and each has an accordion on it with 5-7 items, the experience would be weirdly inconsistent for a user. 

@alexstine Do you have an opinion here? 

### Step-by-step reproduction instructions

1. Create an accordion group.
2. Look at the content and observe the presence of `aria-labelledby` and the absence of any `role` attribute.

### Environment info

- WordPress Trunk
- Gutenberg 21.7.0 RC1

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by alexstine** [comment]
@joedolson I'm fine to use `role="region"` here, but it needs to be done for all.

**Comment by joedolson** [comment]
Thanks for confirming @alexstine! I expected that the inconsistency would be a bad plan, whatever the WAI might recommend.

So, based on that feedback, this is a pretty easy fix: need to add `role="region"` on the accordion panel content in all cases.

**Comment by t-hamano** [comment]
Thanks both of you for your feedback!

> So, based on that feedback, this is a pretty easy fix: need to add `role="region"` on the accordion panel content in all cases.

Let's move forward with this.
